flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -198,10 +211,47 @@ else if (message instanceof SubmitJobAndWait) {
 					decorateMessage(new Status.Failure(new Exception(msg))), ActorRef.noSender());
 			}
 		}
+		else if (message instanceof AttachToJobAndWait) {
```

Comment:
I agree that this design could be improved. I'll consider factoring out common code into a base class.